src/software/ai/hl/stp/tactic/penalty_kick_tactic/penalty_kick_tactic_fsm.h

```diff
@@ -0,0 +1,299 @@
+                current_robot_position = event.common.robot.position();
+                DribbleFSM::ControlParams control_params{
+                    .dribble_destination = std::optional<Point>(next_robot_position),
+                    .final_dribble_orientation = std::nullopt,
+                    .allow_excessive_dribbling = false,
+                };
+                processEvent(DribbleFSM::Update(control_params, event.common));
+            };
+
+        /**
+         * Action that orients the shooter to prepare for a shot.
+         *
+         * @param event          PenaltyKickTacticFSM::Update
+         * @param processEvent   processes the DribbleFSM::Update
+         */
+        const auto adjust_orientation_for_shot =
+            [this](auto event, back::process<DribbleFSM::Update> processEvent) {
+                const Point next_shot_position = evaluateNextShotPosition(
+                    event.control_params.enemy_goalie, event.common.world.field());
+                const Point final_position = event.common.robot.position();
+                shot_angle = (next_shot_position - final_position).orientation();
+                DribbleFSM::ControlParams control_params{
+                    .dribble_destination       = std::nullopt,
+                    .final_dribble_orientation = std::optional<Angle>(shot_angle),
+                    .allow_excessive_dribbling = false,
+                };
+                processEvent(DribbleFSM::Update(control_params, event.common));
+            };
+
+        /**
+         * Guard that returns true if the shooter has a good shot on goal or if it is
+         * forced to shoot due to the penalty timeout.
+         *
+         * @param event  PenaltyKickTacticFSM::Update
+         */
+        const auto take_penalty_shot = [this](auto event) {
+            Timestamp force_shoot_timestamp =
+                complete_approach + PENALTY_FORCE_SHOOT_TIMEOUT;
+            bool shouldShoot =
+                evaluatePenaltyShot(event.control_params.enemy_goalie,
+                                    event.common.world.field(), event.common.world.ball(),
+                                    event.common.robot) ||
+                event.common.world.getMostRecentTimestamp() >= force_shoot_timestamp;
+            return shouldShoot;
+        };
+
+        /**
+         * Guard that checks whether the shooter shot the ball with a particular shot
+         * speed and orientation.
+         *
+         * @param event  PenaltyKickTacticFSM::Update
+         */
+        const auto ball_kicked = [](auto event) {
+            return event.common.world.ball().hasBallBeenKicked(shot_angle,
+                                                               PENALTY_KICK_SHOT_SPEED);
+        };
+
+        return make_transition_table(
+            // src_state + event [guard] / action = dest state
+            *initial_s + update_e / initialize_start_keeper = approach_keeper_s,
```

Comment:
I think you can start the FSM in the approach_keeper_s state

Reply:
done, now some of the shared variables are initialized using a std::optional param